deploy-service/teletraanservice/src/main/java/com/pinterest/teletraan/security/TeletraanAuthZResourceExtractorFactory.java

```diff
@@ -23,51 +23,58 @@ public class TeletraanAuthZResourceExtractorFactory implements AuthZResourceExtr
 
     private static final AuthZResourceExtractor BUILD_BODY_EXTRACTOR = new BuildBodyExtractor();
     private static final AuthZResourceExtractor ENV_PATH_EXTRACTOR = new EnvPathExtractor();
-    private static final AuthZResourceExtractor ENV_STAGE_PATH_EXTRACTOR = new EnvStagePathExtractor();
-    private final AuthZResourceExtractor BUILD_PATH_EXTRACTOR;
-    private final AuthZResourceExtractor ENV_STAGE_BODY_EXTRACTOR;
+    private static final AuthZResourceExtractor ENV_STAGE_BODY_EXTRACTOR =
+            new EnvStageBodyExtractor();
+    private static final AuthZResourceExtractor ENV_STAGE_PATH_EXTRACTOR =
+            new EnvStagePathExtractor();
```

Comment:
What is the difference between these and 
```
  buildPathExtractor = new BuildPathExtractor(serviceContext);
  deployPathExtractor = new DeployPathExtractor(serviceContext);
  hotfixPathExtractor = new HotfixPathExtractor(serviceContext);
```

Reply:
These are simple classes that don't need DBDAO in their constructor.